Adds additional Identify functionality around hidden attributes

davidklaw opened the pull request:
This PR addresses issue #702, where Identify wasn’t revealing item name and description as expected. To address this, I added code to Identify that will temporarily reveal an items hidden attributes, then hide them once again. I took this a step further by also revealing and hidden an items hidden color, if it has one. Examples of this include many of the magical rings.

With this fix, Identify now works as intended _and_ will (temporarily) reveal an item's hidden color.

1. Screenshot of an item with hidden attributes, including color:
![image](https://github.com/user-attachments/assets/f3abae49-d32a-45f5-ab8d-c998641c4779)

2. The same item when Identify is cast upon it, revealing it's true name, true description, and hidden color, but only temporarily in the dialog:
![image](https://github.com/user-attachments/assets/e14c9268-0d53-4f41-bb89-34cf74908c1e)

3. If the user inspects the item again, any attributes that were hidden will be hidden once again:
![image](https://github.com/user-attachments/assets/5a3f66bf-6ba0-495f-a2f6-4490adec7583)

4. And if a user casts Reveal, these changes become permanent:
![image](https://github.com/user-attachments/assets/7a934967-b4b8-414e-92d3-af2579a1328c)

This fixes #702

davidklaw commented:
My proposed implementation is complete and this is ready for review.

davidklaw commented:
@adrienlaws Tagging you since you're familiar with this area of the code.

jrotunda commented:
Must admit, this issue is so minor it was hard to see the problem even when broken down, but do understand it now and it's a good catch for consistency.
For anyone else like me: Reveal permanently removes a shroud, Identify only temporarily - but when showing the temporary item, it was still missing some information, including the color of the revealed ring. 

With the fix, Identify is working as it should. This should also now work with all other items using the Shroud/Hidden Item Atttribute.

davidklaw commented:
I’ve updated the description to better explain what was changed and how. Let me know if anything needs more clarity or if we should scale this back to just fix the original issue. I added the color reveal since it felt like a related improvement, but I’m open to dialing that back if needed.

davidklaw commented:
@Meridian59 Any concerns with this one? If needed, I can split this into two parts:

1. Fixing #702
2. Enhancing with the addition of a temporary color reveal

Let me know how you’d prefer to proceed.

akirmse commented on kod/object/item.kod:
```diff
@@ -1253,6 +1253,23 @@ messages:
+         }
+      }
+      
+      return bDone;
```
This return value is not documented (neither is the function), and appears to have no effect.

davidklaw replied:
Not that this is valid documentation (and I’ll remove it), but this function serves the opposite purpose of the `RevealHiddenAttributes` function just above it. I hadn’t considered that the return value might not be necessary here. Thinking it through, the return value seems unnecessary for this function’s purpose and appears to only be passed by the reveal function to handle a few unrelated tasks.

akirmse commented on kod/object/item.kod:
```diff
@@ -1797,5 +1814,18 @@ messages:
+         Send(self,@SetPaletteTranslation,#translation=viUnrevealedColor);
+         return TRUE;
+      }
+      return FALSE;
```
Return value has no effect?

davidklaw commented:
I've addressed the two comments, but also refactored `Identify::CastSpell` for greater clarity.

davidklaw commented on kod/object/passive/spell/reveal.kod:
```diff
@@ -119,10 +119,9 @@ messages:
+      % Show newly revealed attributes
       if piAdvance
       {      
-         Send(oTarget,@RevealHiddenColor);
```
Note: This is removed because my PR calls `RevealHiddenColor()` from `RevealHiddenAttributes()`.

davidklaw commented on kod/object/item/passitem/ring/lethring.kod:
```diff
@@ -118,7 +118,6 @@ messages:
       % at this point, it's pretty obvious it's not helping them.
       if Send(self, @RevealHiddenAttributes)
       {
-         Send(self, @RevealHiddenColor);
```
Note: This is removed because my PR calls RevealHiddenColor() from RevealHiddenAttributes().
